lib/segment/src/index/field_index/map_index.rs

```diff
@@ -194,14 +323,25 @@ impl PayloadFieldIndex for MapIndex<String> {
         MapIndex::flusher(self)
     }
 
-    fn filter(
-        &self,
-        condition: &FieldCondition,
-    ) -> Option<Box<dyn Iterator<Item = PointOffsetType> + '_>> {
+    fn filter<'a>(
+        &'a self,
+        condition: &'a FieldCondition,
+    ) -> Option<Box<dyn Iterator<Item = PointOffsetType> + 'a>> {
         match &condition.r#match {
             Some(Match::Value(MatchValue {
                 value: ValueVariants::Keyword(keyword),
             })) => Some(self.get_iterator(keyword)),
+            Some(Match::Any(MatchAny {
+                any: AnyVariants::Keywords(keywords),
+            })) => Some(Box::new(
+                keywords
+                    .iter()
+                    .flat_map(move |keyword| self.get_iterator(keyword))
```

Comment:
`move` is not required here:

```suggestion
                    .flat_map(|keyword| self.get_iterator(keyword))
```